Runtime/Properties/SnapZoneProperty.cs

```diff
@@ -62,23 +70,90 @@ public SnapZone SnapZone
         }
 
         private SnapZone snapZone;
-        
+
+        protected virtual void Reset()
+        {
+            if (SnapZone.attachTransform)
+            {
+                return;
+            }
+
+            GameObject attach = new GameObject("Attach Point");
+            attach.transform.SetParent(transform, false);
+            attach.transform.localRotation = Quaternion.identity;
+            attach.transform.localPosition = Vector3.zero;
+            attach.transform.localScale = Vector3.one;
+
+            FieldInfo attachField = typeof(XRBaseInteractor).GetField("m_AttachTransform", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (attachField != null)
+            {
+                attachField.SetValue(SnapZone, attach.transform);
+            }
+            else
+            {
+                DestroyImmediate(attach);
+
+                // It seems that the Unity team has renamed that field. Check the SnapZone script. 
+                Debug.Log("We failed to add an attach point for this snap zone property. Please, drop us a word if you see this message.");
+            }
+
+            List<Collider> colliders = GetComponentsInChildren<Collider>().Where(foundCollider => foundCollider.isTrigger).ToList();
+
+            if (colliders.Count == 0)
+            {
+                Debug.Log($"{addAttachAtOrigin} {tooLittleCollidersError} {attachPointExplanation}");
+                Debug.LogWarning("This snap zone has no trigger colliders. It will not snap any objects, and will physically block them. Add a collider to the snap zone object if it has none. Check the Is Triggered box of its collider.");
+                return;
+            }
+            
+            if (colliders.Count > 1)
+            {
+                Debug.Log($"{addAttachAtOrigin} {tooManyCollidersError} {attachPointExplanation}");
+                return;
+            }
```

Comment:
We could use [`Bounds.Encapsulate `](https://docs.unity3d.com/ScriptReference/Bounds.Encapsulate.html) to obtain the center of all valid colliders.

Reply:
It feels like a solution that is too smart and magical to be good.

Reply:
I know, I love it too, take a look at https://github.com/Innoactive/XR-Interaction-Component/blob/cbf8f92c2c9c4cda4026eded526b2571cdfbcb0f/Editor/Properties/SnappablePropertyEditor.cs#L50-L60

Reply:
Yeah, but mesh collider can have an offset only if all its points have an offset. and if you you usethe same mesh in renderer for the snapped object, it will have the same offset. So making a weird offset by the center of AABB will actually break the correst visuals.

Reply:
No, I am confused.

Reply:
What if you use the center of the Snap Zone's Box Collider, it supposed to already be centered according to the renderer and not other colliders (like a mesh collider).

Reply:
Yes, but it happens already. We create a box collider with snap zone wizard, then we add the SZP, SZP.Reset() finds the box collider and sets the offset to its center.